File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {
        
        int n=nums.size();
        int res=0,cnt=0;
        for(int i=0;i<(1<<n);i++)
        {
            int ans=0;
            for(int j=0;j<n;j++)
            {
                 if ((i & (1 << j)) != 0) {
                    //cout<<nums[j]<<" ";
                    ans|=nums[j];
                 }
            }
            //cout<<"hj";
            if(res<ans)
            {
                res=ans;
                cnt=1;
            }
            else if(res==ans){
                cnt++;
            }
        }
        return cnt;
    }
};

File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.py
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        max_or = 0
        prev = Counter()
        prev[0] = 1
        
        for elem in nums:
            max_or |= elem

            current = Counter()
            for prev_or, cnt in prev.items():
                current[prev_or | elem] += cnt
            prev.update(current)
        
        return prev[max_or]
        